skbio/sequence/_sequence.py

```diff
@@ -111,98 +135,83 @@ def values(self):
-    def quality(self):
-        """Quality scores corresponding to biological sequence characters.
-
-        A 1D ``np.ndarray`` of nonnegative integers representing a Phred
-        quality score for each character in the biological sequence, or
-        ``None`` if quality scores are not present.
+    def positional_metadata(self):
+        """Pandas DataFrame containing metadata on a per-character basis.
 
         Notes
         -----
-        This property is not writeable. A copy of the array is *not* returned.
-        The array is read-only (i.e., its ``WRITEABLE`` flag is set to
-        ``False``).
+        This property is writeable.
 
         Examples
         --------
         >>> from skbio import Sequence
-        >>> s = Sequence('GGUCGUGACCGA', id='seq1', description='some seq',
-        ...              quality=[1, 5, 3, 3, 2, 42, 100, 9, 10, 0, 42, 42])
-        >>> s.quality
-        array([  1,   5,   3,   3,   2,  42, 100,   9,  10,   0,  42,  42])
+        >>> s = Sequence('ACGTACGTACGTACGT',
+        ...              positional_metadata={'quality': range(16)})
+        >>> s # doctest: +NORMALIZE_WHITESPACE
+        Sequence('ACGTACGTACGTACGT', length=16, has_metadata=False,
+                 has_positional_metadata=True)
+        >>> s.positional_metadata['quality'].values
+        array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15])
 
         """
-        return self._quality
+        return self._positional_metadata
 
     @property
     def _string(self):
         return self._bytes.tostring()
 
-    def __init__(self, sequence, id="", description="", quality=None):
+    def __init__(self, sequence, metadata=None,
+                 positional_metadata=None):
         if isinstance(sequence, Sequence):
-            if id == "":
-                id = sequence.id
-            if description == "":
-                description = sequence.description
-            if quality is None:
-                quality = sequence.quality
+            if metadata is None:
+                metadata = sequence.metadata
+            if positional_metadata is None:
+                positional_metadata = sequence.positional_metadata
 
             sequence = sequence._bytes
 
-        self._set_id(id)
-        self._set_description(description)
-        self._set_sequence(sequence)
-        self._set_quality(quality)
+        if metadata is None:
+            metadata = {}
+        elif not isinstance(metadata, dict):
+            raise TypeError("metadata must be a {}".format(type(dict())))
+        self._metadata = metadata
 
-    def _set_id(self, id_):
-        if isinstance(id_, str):
-            self._id = id_
-        else:
-            raise TypeError("Sequence ID %r must be of type `str`." % (id_,))
+        try:
+            self._positional_metadata = pd.DataFrame(positional_metadata)
+        except pd.core.common.PandasError:
+            raise TypeError("Positional metadata invalid. Must be consumable "
```

Comment:
hrmmm.... can you give an example of an invalid input that would trigger this error? and the error message that pandas provides